scripts/_helpers.py

```diff
@@ -966,33 +966,74 @@ def annuity(n, r):
         return 1 / n
 
 
+# Simple cache to avoid repeated computations and logging for same (currency, output_currency, year)
+_conversion_cache = {}
+
+
 def get_yearly_currency_exchange_average(
```

Comment:
It is worth having _conversion_cache as argument of the function to avoid the hidden dependency and allow customization of entries potentially. The name of the variable may be revised in _currency_conversion_cache to make it more explicit maybe

Reply:
It is worth adding a docstring explaining the function